test/integration/interface_api.cc

```diff
@@ -397,26 +399,53 @@ TEST_F(InterfaceAPI, TomlParserModelInclude)
   TomlParserTest(interfaceModel);
 }
 
+/////////////////////////////////////////////////
+TEST_F(InterfaceAPI, DeeplyNestedModel)
+{
+  const std::string testSdf = R"(
+  <sdf version="1.10">
+    <model name="grand_parent_model">
+      <include>
+        <uri>model://model_include_with_interface_api.sdf</uri>
+      </include>
+    </model>
+  </sdf>)";
+
+  this->config.AddURIPath("model://", sdf::testing::TestFile("sdf"));
+  this->config.RegisterCustomModelParser(this->customTomlParser);
+  sdf::Root root;
+  sdf::Errors errors = root.LoadSdfString(testSdf, this->config);
+  EXPECT_TRUE(errors.empty()) << errors;
+  const sdf::Model *grandParentModel = root.Model();
+  ASSERT_NE(nullptr, grandParentModel);
+  const sdf::Model *parentModel = grandParentModel->ModelByIndex(0);
+  ASSERT_NE(nullptr, parentModel);
```

Comment:
nit: add `EXPECT_EQ("parent_model", parentModel->Name());`

Reply:
Done in https://github.com/gazebosim/sdformat/pull/1293/commits/e852813155496fa72b3d4d9b530f1200a5195142